### PR TITLE
Merge appsembler/ficus/develop to appsembler/ficus/master branch 

### PR DESCRIPTION
**Pull request by OmarIthawi**
This is mainly to get the #184 released to ASU staging and production. The API has been deployed for a while. I've also created a PR (https://github.com/appsembler/edx-configs/pull/226) to get the ASU Staging back to `appsembler/ficus/master`.

### CODE REVIEW

**Comment by OmarIthawi** [comment]
[(8) ASU Starbucks: Add a search API for mobile](https://trello.com/c/l3YpwqhJ/2355-8-asu-starbucks-add-a-search-api-for-mobile)

**Comment by OmarIthawi** [comment]
@bryanlandia @amirtds @tkeemon could you please take a look?

File: lms/envs/devstack_appsembler.py [hunk 1]
```diff
@@ -122,3 +122,5 @@
     pass
 
 
+# override devstack.py automatic enabling of courseware discovery
+FEATURES['ENABLE_COURSE_DISCOVERY'] = ENV_TOKENS.get('TPA_CLEAN_USERNAMES_KEEP_DOMAIN_PART', False)
```
Comment (melvinsoft) [inline lms/envs/devstack_appsembler.py]: I think you calling a wrong setting here: `TPA_CLEAN_USERNAMES_KEEP_DOMAIN_PART `

**Comment by OmarIthawi** [comment]
Same issue, @bryanlandia could you please check this out?

File: lms/envs/devstack_appsembler.py (hunk 1 above)
Comment (bryanlandia) [reply lms/envs/devstack_appsembler.py]: @OmarIthawi - Yes, this doesn't look right.  Looks like that was my commit—not sure how that got so wrong... Should be 

```
FEATURES['ENABLE_COURSE_DISCOVERY'] = ENV_TOKENS['FEATURES'].get('ENABLE_COURSE_DISCOVERY', False)
```

This is overriding a line in `devstack.py` which overrides the original `FEATURE` definition to always turns on course discovery.

**Comment by OmarIthawi** [comment]
@melvinsoft We've addressed the issue. I'll merge.

**Pull request merged**